mne/io/eeglab/tests/test_eeglab.py

```diff
@@ -35,8 +36,28 @@
 
 
 @testing.requires_testing_data
+@requires_h5py
 def test_io_set():
     """Test importing EEGLAB .set files."""
+    _test_io_set(False)
+    _test_io_set(True)
+
+
+@testing.requires_testing_data
+@requires_h5py
+def _test_io_set(use_hdf=True):
+    """Test importing EEGLAB .set files."""
+    global raw_fname
+    global raw_fname_onefile
+    global epochs_fname
+    global epochs_fname_onefile
```

Comment:
Avoid using `global` where possible. Call these filenames `epochs_fname_h5` etc. at the top with the others, then triage here which one to load.

Reply:
I incorporated the structural suggestions. Tests pass on my machine, but we'll see what happens when I push.

I undid the change to base.py, because I can't recreate the error that necessitated it (which is weird, but so was finding something that needed to be fixed in base.py). The weird part of the error was that I couldn't figure out how a rewEEGLAB object managed to avoid having a filename attribute assigned in the first place.

I removed the globals, but am still not certain I should have. I was using the globals to overshadow some module wide variables within a local method, without unduly effecting the other methods in the modules, in this case, since it's a unit test module, but maybe I'm wrong.